core/src/ldml/ldml_processor.cpp

```diff
@@ -298,19 +298,26 @@ size_t ldml_processor::process_output(ldml_event_state &ldml_state, const std::u
   //  The normalize functions have assert and Debuglog at the bottom.
   //  so we do not need to assert the status here unless we're going to do something
   //  different with control flow.
-  (void)ldml::normalize_nfd_markers(nfd_str);
-
+  if (!normalization_disabled) {
+    auto norm_ok = ldml::normalize_nfd_markers(nfd_str);
+    assert(norm_ok);
```

Comment:
```suggestion
    assert(ldml::normalize_nfd_markers(nfd_str));
```

Can we do this and reduce the noise?

Reply:
@mcdurdin except that the `assert()` is not called in production, right? maybe we need a new construct.  

`#define assert(x)` is the usual idiom for production mode.

Reply:
Ah of course. My bad ... that could be ugly.

`assert(x)` is dangerous like that... That does mean that `norm_ok` is declared and unused in production code which is possibly a separate painful issue.

Reply:
if the code was extracted out, then the pain would be only in one place.

But actually.  the inner level `normalize()` function *already* asserts.  So one thing I could do is just take out all of these other asserts.

Reply:
hah, and i had added this comment three weeks ago:

```
size_t ldml_processor::process_output(ldml_event_state &ldml_state, const std::u32string &str, ldml::transforms *with_transforms) const {
  std::u32string nfd_str = str;
  // Note:
  //  The normalize functions have assert and Debuglog at the bottom.
  //  so we do not need to assert the status here unless we're going to do something
  //  different with control flow.
```